src/riot-rs-rp/src/storage.rs

```diff
@@ -0,0 +1,14 @@
+use embassy_rp::flash::Async;
+use embassy_rp::peripherals::FLASH;
+
+pub type Flash = embassy_rp::flash::Flash<'static, FLASH, Async, FLASH_SIZE>;
+pub type FlashError = embassy_rp::flash::Error;
+
+const FLASH_SIZE: usize = 2 * 1024 * 1024;
```

Comment:
Every rp2040 comes with flash that's accessible through this interface, otherwise it won't run. Though, maybe the size won't be `>=2mb` for all of them. Can we make this more flexible in the next PRs, that add more flexbilities w.r.t. board definitions?
Also, add rp2350 support in #435?